pkg/cluster/defaults.go

```diff
@@ -7,19 +7,37 @@ import (
 	"context"
 
 	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database"
 )
 
 // ensureDefaults will ensure cluster documents has all default values
 // for new api versions
 func (m *manager) ensureDefaults(ctx context.Context) error {
-	var err error
-	m.doc, err = m.db.PatchWithLease(ctx, m.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
+	return patchClusterDoc(ctx, m, func(doc *api.OpenShiftClusterDocument) error {
 		api.SetDefaults(doc)
 		return nil
 	})
+}
+
+// ensureBYONsg updates the cluster docs to disable operators that might interfere
+// with BYO NSG feature
+func (m *manager) ensureBYONsg(ctx context.Context) error {
+	if m.byoNSG {
+		return patchClusterDoc(ctx, m, func(doc *api.OpenShiftClusterDocument) error {
+			flags := doc.OpenShiftCluster.Properties.OperatorFlags
+			flags["aro.azuresubnets.nsg.managed"] = "false"
+			flags["aro.azuresubnets.enabled"] = "false"
```

Comment:
is there a reason we're setting subnets to disabled as well?  I think subnets should still be setting service endpoints.